checksum.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2012 Google, Inc. All rights reserved.
+//
+// Use of this source code is governed by a BSD-style license
+// that can be found in the LICENSE file in the root of the source
+// tree.
+
+package gopacket
+
```

Comment:
uint32 as a primitive type for checksum would become confusing. let's add a Checksum type here and convert the `Correct` and `Actual` types to it.